vernac/declare.ml

```diff
@@ -1404,8 +1404,10 @@ let initialize_named_context_for_proof () =
   List.fold_right
     (fun d signv ->
       let id = NamedDecl.get_id d in
-      let d = if Decls.variable_opacity id then NamedDecl.drop_body d else d in
-      Environ.push_named_context_val d signv) sign Environ.empty_named_context_val
+      if Decls.variable_goal_visibility id then
+        let d = if Decls.variable_opacity id then NamedDecl.drop_body d else d in
+        Environ.push_named_context_val d signv
+      else signv) sign Environ.empty_named_context_val
```

Comment:
If we don't put ring's lemmas in the context ring can't use them, so I'm not sure how it's supposed to work.